pmm/summary.go

```diff
@@ -0,0 +1,263 @@
+		{"Collect list of open ports",
+		[]string{"netstat", "-punta"},
+		strings.Join([]string{dirname,strings.Join([]string{"netstat_",string(cmdHostname)[:len(string(cmdHostname))-1],".txt"},"")},"/")}}
+
+	initType := CheckInitType()
+	switch {
+		case (initType == "init"):
+			Collectors = append(Collectors, Collector{"Collect service output",
+			[]string{"service", "--status-all"},
+			strings.Join([]string{dirname,strings.Join([]string{"sysv_service_",string(cmdHostname)[:len(string(cmdHostname))-1],".txt"},"")},"/")})
+		case (initType == "systemd"):
+			Collectors = append(Collectors, Collector{"Collect systemctl output",
+			[]string{"systemctl", "-l", "status"},
+			strings.Join([]string{dirname,strings.Join([]string{"systemd_",string(cmdHostname)[:len(string(cmdHostname))-1],".txt"},"")},"/")})
+		case (initType == "upstart"):
+			Collectors = append(Collectors, Collector{"Collect initctl output",
+			[]string{"initctl","list"},
+			strings.Join([]string{dirname,strings.Join([]string{"upstart_",string(cmdHostname)[:len(string(cmdHostname))-1],".txt"},"")},"/")})
+	}
+
+	for _, service := range CheckMonitoredDBServices() {
+		switch {
+			case (service == "mysql"):
+				if (len(mysqlsocket)>0) { mysqlsocket = strings.Join([]string{"--socket=",mysqlsocket},"")}
+				if (len(mysqluser)>0) { mysqluser = strings.Join([]string{"--user=",mysqluser},"")}
+				if (len(mysqlpassword)>0) { mysqlpassword = strings.Join([]string{"--password=",mysqlpassword},"")}
+				if (len(mysqlport)>0) { mysqlport = strings.Join([]string{"--port=",mysqlport},"")}
+				Collectors = append(Collectors, Collector{"Collect pt-mysql-summary output",
+				[]string{"pt-mysql-summary",mysqluser,mysqlpassword,mysqlsocket,mysqlport},
+				strings.Join([]string{dirname,strings.Join([]string{"pt-mysql-summary_",string(cmdHostname)[:len(string(cmdHostname))-1],".txt"},"")},"/")})
+			case (service == "mongodb"):
+				Collectors = append(Collectors, Collector{"Collect pt-mongodb-summary output",
+				[]string{"pt-mongodb-summary"},
+				strings.Join([]string{dirname,strings.Join([]string{"pt-mongodb-summary_",string(cmdHostname)[:len(string(cmdHostname))-1],".txt"},"")},"/")})
+		}
+	}
+
+	// Perform exec commands and collect results
+	for _, info := range Collectors {
+		info.CollectData()
+	}
+
+	// Collect pmm logs from /var/log
+	pmmLogs, _ := regexp.Compile("^pmm-.*")
+	srcf, err := os.Open("/var/log");
+	if err != nil {
+		fmt.Printf("Failed to open directory %s with %s\n", dirname, err)
+	} else {
+		defer srcf.Close()
+		names, err := srcf.Readdirnames(100)
+		if err != nil {
+			fmt.Printf("Failed to read directory %s with %v\n", dirname, err)
+		} else {
+			for _, name := range names {
+				if pmmLogs.MatchString(name) {
+					srcf, err := os.Open(strings.Join([]string{"/var/log/",name},"/"))
+					if err != nil {
+						fmt.Printf("Open file %s failed with %v\n", strings.Join([]string{"/var/log/",name},"/"), err)
```

Comment:
@AlekSi what do you think about those errors not being bubbled up?
Also maybe separate `.err` file would be best option for handling errors in a log zip file?
```
prometheus.log // if it exist
prometheus.err // if it doesn't exit, or there were some additional errors then we can stream them here too
```

Reply:
It was imported the log package and errors are stored into file pmm-summary.err